update-tests/src/test/java/io/zeebe/test/ContainerState.java

```diff
@@ -18,15 +18,16 @@
 import java.util.regex.Pattern;
 import net.jodah.failsafe.Failsafe;
 import net.jodah.failsafe.RetryPolicy;
+import org.junit.jupiter.api.extension.ExtensionContext.Store.CloseableResource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testcontainers.containers.ContainerFetchException;
 import org.testcontainers.containers.ContainerLaunchException;
 import org.testcontainers.containers.Network;
 
-public class ContainerState {
+final class ContainerState implements CloseableResource {
```

Comment:
In junit5, `CloseableResource` instances which are stored in an extension store will be automatically close at the end of the extension's lifecycle - in our case, with `ContainerStateExtension`, this means at the end of the test where the parameter is injected.